runelite-client/src/main/java/net/runelite/client/plugins/twitch/TwitchPlugin.java

```diff
@@ -0,0 +1,420 @@
+		{
+			username = "<img=26>" + username;
+		}
+
+		//Streamer badge
+		if (a.getOriginalMessages().get(0).getTag("badges").toString().contains("broadcaster/1"))
+		{
+			username = "<img=27>" + username;
+		}
+
+		//Show the message in game with color based on if the chat is transparent or not
+		String message = a.getMessage();
+		if (isChatboxTransparent() && client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+					"<col=ffffff>[</col><col=9070ff>Twitch</col>] " + username + ": <col=" + Integer.toHexString(config.chatTColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+		}
+		else if (client.getGameState().equals(GameState.LOGGED_IN))
+		{
+			client.addChatMessage(ChatMessageType.TRADE, "",
+					"<col=000000>[</col><col=0000FF>Twitch</col>] " + username + ": <col=" + Integer.toHexString(config.chatOColor().getRGB() & 0xFFFFFF) + ">" + message + "</col>", "Twitch");
+		}
+	}
+
+	//User and room information update events
+	//Triggers whenever the user's state is updated or they join a room
+	@Handler
+	public void onUserStateEvent(UserStateEvent u)
+	{
+		String originalMessage = u.getOriginalMessage().toString();
+		badges = "";
+
+		//Finding if the user is a subscriber
+		if ((originalMessage.contains("subscriber=1") || originalMessage.contains("mod=1") || originalMessage.contains("value=broadcaster/1")) && StringUtils.containsIgnoreCase(originalMessage, "display-name=" + config.channelName()))
+		{
+			subscribed = true;
+			if (originalMessage.contains("subscriber=1"))
+			{
+				if (originalMessage.contains("subscriber/0"))
+				{
+					badges = "<img=9>" + badges;
+				}
+				else if (originalMessage.contains("subscriber/3"))
+				{
+					badges = "<img=8>" + badges;
+				}
+				else if (originalMessage.contains("subscriber/6"))
+				{
+					badges = "<img=7>" + badges;
+				}
+				else if (originalMessage.contains("subscriber/12"))
+				{
+					badges = "<img=6>" + badges;
+				}
+				else if (originalMessage.contains("subscriber/24"))
+				{
+					badges = "<img=5>" + badges;
+				}
+				else if (originalMessage.contains("subscriber/48"))
+				{
+					badges = "<img=4>" + badges;
+				}
+			}
+		}
+		else
+		{
+			subscribed = false;
+		}
+
+		//Finding if the user is a moderator or broadcaster which allows them to talk during emote-only
+		if (originalMessage.contains("mod=1") && StringUtils.containsIgnoreCase(originalMessage, "display-name=" + config.channelName()))
+		{
+			mod = true;
+			badges = "<img=26>" + badges;
+		}
+		else
+		{
+			mod = false;
+		}
+
+		if (originalMessage.contains(("value=broadcaster/1")) && StringUtils.containsIgnoreCase(originalMessage, "display-name=" + config.channelName()))
+		{
+			broadcaster = true;
+			badges = "<img=27>" + badges;
+		}
+		else
+		{
+			broadcaster = false;
+		}
+	}
+
+	//Triggers when the room state changes. I.E. Sub mode, slow mode, etc.
```

Comment:
Change to doc comment

Reply:
The library I'm using already has javadocs and I've not done doc comments before, should I just use what's in their docs? http://kittehorg.github.io/KittehIRCClientLib/

Reply:
Or I guess I just change it to this 
![image](https://user-images.githubusercontent.com/39203754/40880439-28ee5b1a-6687-11e8-8910-dda379e41090.png)

Reply:
Yep, like that. Put a space between the * and T ;)